contracts/governance/TimelockController.sol

```diff
@@ -260,8 +302,13 @@ contract TimelockController is AccessControl, IERC721Receiver, IERC1155Receiver
      * @dev Schedule an operation that is to become valid after a given delay.
      */
     function _schedule(bytes32 id, uint256 delay) private {
-        require(!isOperation(id), "TimelockController: operation already scheduled");
-        require(delay >= getMinDelay(), "TimelockController: insufficient delay");
+        if (isOperation(id)) {
+            revert TimelockDuplicatedOperation(id);
```

Comment:
We could do

```suggestion
            revert TimelockIncorrectState(id, OperationState.Unset);
```

Reply:
This is a very nice idea, I just implemented it 🎉